### DISCUSSION TITLE
VERY Brief Question - Multiple Rules, Same Class

### DISCUSSION THREAD

**Discussion by justinstroud-campmeeker** [pmd/pmd#5194] [Q&A] [answered]
I have two rules that point to the same Java class (this is a custom rule I built). Each rule has a different name and different property values in the ruleset XML. It doesn't seem to work.

How can I have two rules with different names, different property values, but pointing to the same class? Similar to XPath rules.

Thank you

**Comment by jsotuyod** [comment]
It should work… care to share the ruleset xml?
Just to be clear, what's the observed behavior? None of the rules work? Only one?
What's the output when running PMD in verbose mode? (`-v` in the CLI, `-X` with Maven, `-d` in Gradle)

**Reply by justinstroud-campmeeker** [reply]
Only one. Let me do a quick test and I'll share that with you.

**Reply by justinstroud-campmeeker** [reply, marked as answer]
@jsotuyod I fixed the issue. It was giving me that vague "XML validation...can't find the rule" error and it wasn't the XML...it was a formatting error of the property in my ruleset. I've had this error before, and it was a bit of a red herring. Attaching my working rules for posterity. (took me a bit, had to sanitize my rules)
[scripted.zip](https://github.com/user-attachments/files/16910433/scripted.zip)
